### DISCUSSION TITLE
"Absolute mode" in Craft.js

### DISCUSSION THREAD

**Discussion by charlyv** [prevwong/craft.js#457] [Ideas]
Hi there,

We are looking at doing a POC of Craft.js where we experiment with enabling/extending it with an "absolute" mode, and thus validate that we could use a DOM based approach to emulate what can normally be done more natively with the HTML Canvas element, in addition to the default and current editing mode of Craft.js.

This would enable us to build a feature suite on top of Craft.js that enables our users to create either web pages using the standard "DOM" mode, or static or animated designs and presentations using the "absolute" mode (e.g. like Canva, Create Vista, Adobe Express...), using the same framework, which would create efficiencies in implementation, since the building experience around it, and the widgets/blocks and their properties will be mostly the same across both modes.

Has anyone else here experimented with enabling Canvas like interactions in Craft.js?

**Comment by ankri** [comment]
There are some discussions and issues regarding absolute positioning. There are also some PoCs you can copy the code from.

**Reply by charlyv** [reply]
Ok cool, will try again to find those

**Reply by charlyv** [reply]
Any link you can share would be greatly appreciated as searching for "absolute" only returns that post. Thanks in advance

**Reply by ankri** [reply]
https://github.com/prevwong/craft.js/issues?q=absolute

**Reply by charlyv** [reply]
Thanks a lot @ankri - my bad, I was only searching in discussions. This is great info
